src/microbe_stage/Microbe.Interior.cs

```diff
@@ -389,38 +389,34 @@ public Microbe Divide()
 
         var currentPosition = GlobalTransform.origin;
 
-        // Separate the two cells.
-        var separation = new Vector3(Radius, 0, 0);
+        // Find the direction to the right from where the cell is facing
+        var direction = GlobalTransform.basis.Quat().Normalized().Xform(Vector3.Right);
 
-        if (Colony != null)
-        {
-            // When in a colony we approximate a much higher separation distance
-            var colonyRadius = separation.x;
-
-            foreach (var colonyMember in Colony.ColonyMembers)
-            {
-                if (colonyMember == this)
-                    continue;
+        // Start calculating separation distance
+        var organellePositions = organelles!.Organelles.Select(o => Hex.AxialToCartesian(o.Position)).ToList();
 
-                var radius = colonyMember.Radius + Constants.COLONY_DIVIDE_EXTRA_DAUGHTER_OFFSET;
+        float distanceRight = MathUtils.GetMaximumDistanceInDirection(Vector3.Right, Vector3.Zero, organellePositions);
+        float distanceLeft = MathUtils.GetMaximumDistanceInDirection(Vector3.Left, Vector3.Zero, organellePositions);
 
-                // TODO: switch this to something else if this is too slow for large colonies
-                var positionInColony = colonyMember.GlobalTransform.origin - currentPosition;
+        if (Colony != null)
+        {
+            var colonyMembers = Colony.ColonyMembers.Select(c => c.GlobalTransform.origin).ToList();
 
-                var outerRadius = Math.Max(Math.Abs(positionInColony.x) + radius,
-                    Math.Abs(positionInColony.z) + radius);
+            distanceRight += MathUtils.GetMaximumDistanceInDirection(direction, currentPosition, colonyMembers);
+        }
 
-                if (outerRadius > colonyRadius)
-                    colonyRadius = outerRadius;
-            }
+        float width = distanceLeft + distanceRight + Constants.DIVIDE_EXTRA_DAUGHTER_OFFSET;
 
-            separation = new Vector3(colonyRadius + Constants.COLONY_DIVIDE_EXTRA_DAUGHTER_OFFSET, 0, 0);
-        }
+        if (CellTypeProperties.IsBacteria)
+            width *= 0.5f;
 
         // Create the one daughter cell.
-        var copyEntity = SpawnHelpers.SpawnMicrobe(Species, currentPosition + separation,
+        var copyEntity = SpawnHelpers.SpawnMicrobe(Species, currentPosition + direction * width,
             GetParent(), SpawnHelpers.LoadMicrobeScene(), true, cloudSystem!, spawnSystem!, CurrentGame);
 
+        // Since the daughter spawns right next to the cell, it should face the same way to avoid colliding
+        copyEntity.Rotation = Rotation;
```

Comment:
Using Rotation destroys the quaternion by flattening it into euler angles... This should instead modify the transform, something like this (I didn't test so this might be slightly off):

```
copyEntity.Transform = new Transform(Transform.basis, copyEntity.Translation);
```

Reply:
Note that that assumes that the scale etc. is fine in the parent cell to copy. Instead it might be better to convert the `Transform.basis.Quat()` and build a new basis from that that copies the scale from the copyEntity basis.